### PR TITLE
disable telemetry on mojave

### PR DESCRIPTION
**Pull request by MisterTea**

### CODE REVIEW

File: src/base/Headers.hpp [hunk 1]
```diff
@@ -161,14 +163,20 @@ inline int GetErrno() {
...
     switch (retval) {
-    case WSAEWOULDBLOCK: return EWOULDBLOCK;
-    case WSAEADDRINUSE: return EADDRINUSE;
-    case WSAEINPROGRESS: return EINPROGRESS;
-    case WSAENOTSOCK: return ENOTSOCK;
-    case WSAECONNRESET: return ECONNRESET;
-    case WSAECONNABORTED: return ECONNABORTED;
-    default:
-      STFATAL << "Unmapped WSA error: " << retval;
+      case WSAEWOULDBLOCK:
```
Comment (dgentry) [inline src/base/Headers.hpp:166]: Just curious, did clang google format do this?

**Comment by dgentry** [comment]
Builds fine.  `Et` works but `etserver` hangs with a message `Abort 6.`  If only there were some automatic way you could retrieve and examine the crash artifacts.  

I'll have to try further tomorrow.  The port `etserver` was trying to open was already in use by the LaunchDaemon instance of `etserver,` but it didn't get better after I stopped the LaunchDaemon one (and killed the instance I had started by hand).

**Comment by dgentry** [comment]
OK, it compiles, and `et` and `etserver` work, although there's some port contention when another instance `etserver` has been running.  This is on MacOS 10.14.6 with the latest XCode that will run on Mojave.

Here's the build log for posterity:

```
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal $ cd build
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ make -j13
[  3%] Built target generated-code
Consolidate compiler generated dependencies of target et-lib
[ 31%] Built target et-lib
Consolidate compiler generated dependencies of target HtmCommon
Consolidate compiler generated dependencies of target TerminalCommon
[ 60%] Built target HtmCommon
[ 63%] Built target TerminalCommon
Consolidate compiler generated dependencies of target et
Consolidate compiler generated dependencies of target etserver
Consolidate compiler generated dependencies of target htm
Consolidate compiler generated dependencies of target htmd
[ 68%] Built target et
[ 78%] Built target etserver
[ 78%] Built target htm
[ 83%] Built target htmd
Consolidate compiler generated dependencies of target etterminal
[ 88%] Built target etterminal
Consolidate compiler generated dependencies of target et-test
[100%] Built target et-test
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ make clean
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ make -j13
[  3%] Running cpp protocol buffer compiler on proto/ET.proto
[  3%] Running cpp protocol buffer compiler on proto/ETerminal.proto
[  3%] Built target generated-code
[  4%] Generating CXX prefix source cotire/et-lib_CXX_prefix.cxx
[  6%] Generating CXX prefix header cotire/et-lib_CXX_prefix.hxx
[  6%] Building CXX precompiled header cotire/et-lib_CXX_prefix.hxx.pch
[  6%] Building CXX object CMakeFiles/et-lib.dir/external/easyloggingpp/src/easylogging++.cc.o
[  8%] Building CXX object CMakeFiles/et-lib.dir/src/base/BackedReader.cpp.o
[  9%] Building CXX object CMakeFiles/et-lib.dir/external/PlatformFolders/sago/platform_folders.cpp.o
[ 11%] Building CXX object CMakeFiles/et-lib.dir/src/base/BackedWriter.cpp.o
[ 11%] Building CXX object CMakeFiles/et-lib.dir/src/base/ClientConnection.cpp.o
[ 13%] Building CXX object CMakeFiles/et-lib.dir/src/base/CryptoHandler.cpp.o
[ 14%] Building CXX object CMakeFiles/et-lib.dir/src/base/Connection.cpp.o
[ 16%] Building CXX object CMakeFiles/et-lib.dir/src/base/ServerClientConnection.cpp.o
[ 16%] Building CXX object CMakeFiles/et-lib.dir/src/base/ServerConnection.cpp.o
[ 21%] Building CXX object CMakeFiles/et-lib.dir/src/base/SocketHandler.cpp.o
[ 19%] Building CXX object CMakeFiles/et-lib.dir/src/base/TcpSocketHandler.cpp.o
[ 21%] Building CXX object CMakeFiles/et-lib.dir/src/base/PipeSocketHandler.cpp.o
[ 22%] Building CXX object CMakeFiles/et-lib.dir/src/base/UnixSocketHandler.cpp.o
[ 27%] Building CXX object CMakeFiles/et-lib.dir/src/base/DaemonCreator.cpp.o
[ 27%] Building CXX object CMakeFiles/et-lib.dir/src/base/SubprocessToString.cpp.o
[ 27%] Building CXX object CMakeFiles/et-lib.dir/ET.pb.cc.o
[ 27%] Building CXX object CMakeFiles/et-lib.dir/src/base/RawSocketUtils.cpp.o
[ 27%] Building CXX object CMakeFiles/et-lib.dir/src/base/LogHandler.cpp.o
/Volumes/TB/Users/gentry/EternalTerminal/src/base/DaemonCreator.cpp:5:53: warning: 'daemon' is deprecated: first deprecated in macOS 10.5 - Use posix_spawn APIs instead. [-Wdeprecated-declarations]
int DaemonCreator::createSessionLeader() { return ::daemon(0, 0); }
                                                    ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/stdlib.h:292:6: note: 'daemon' has been explicitly marked deprecated here
int      daemon(int, int) __DARWIN_1050(daemon) __OSX_AVAILABLE_BUT_DEPRECATED_MSG(__MAC_10_0, __MAC_10_5, __IPHONE_2_0, __IPHONE_2_0, "Use posix_spawn APIs instead.") __WATCHOS_PROHIBITED __TVOS_PROHIBITED;
         ^
1 warning generated.
[ 29%] Linking CXX static library libet-lib.a
[ 31%] Built target et-lib
[ 32%] Generating CXX prefix source cotire/HtmCommon_CXX_prefix.cxx
[ 34%] Generating CXX prefix source cotire/TerminalCommon_CXX_prefix.cxx
[ 37%] Generating CXX prefix header cotire/HtmCommon_CXX_prefix.hxx
[ 37%] Generating CXX prefix header cotire/TerminalCommon_CXX_prefix.hxx
[ 39%] Building CXX precompiled header cotire/HtmCommon_CXX_prefix.hxx.pch
[ 39%] Building CXX precompiled header cotire/TerminalCommon_CXX_prefix.hxx.pch
[ 42%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/TerminalHandler.cpp.o
[ 40%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/MultiplexerState.cpp.o
[ 42%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/IpcPairClient.cpp.o
[ 45%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/IpcPairEndpoint.cpp.o
[ 45%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/IpcPairServer.cpp.o
[ 45%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/HtmClient.cpp.o
[ 47%] Building CXX object CMakeFiles/HtmCommon.dir/src/htm/HtmServer.cpp.o
[ 49%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/forwarding/PortForwardHandler.cpp.o
[ 49%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/forwarding/ForwardSourceHandler.cpp.o
[ 50%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/forwarding/ForwardDestinationHandler.cpp.o
[ 52%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/TerminalServer.cpp.o
[ 52%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/TerminalClient.cpp.o
[ 54%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/UserTerminalRouter.cpp.o
[ 55%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/SshSetupHandler.cpp.o
[ 57%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/UserTerminalHandler.cpp.o
[ 59%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/UserJumphostHandler.cpp.o
[ 60%] Building CXX object CMakeFiles/TerminalCommon.dir/src/terminal/TelemetryService.cpp.o
[ 60%] Building CXX object CMakeFiles/TerminalCommon.dir/ETerminal.pb.cc.o
[ 62%] Linking CXX static library libHtmCommon.a
[ 62%] Built target HtmCommon
[ 63%] Generating CXX prefix source cotire/htmd_CXX_prefix.cxx
[ 63%] Generating CXX prefix source cotire/htm_CXX_prefix.cxx
[ 65%] Generating CXX prefix header cotire/htmd_CXX_prefix.hxx
[ 67%] Generating CXX prefix header cotire/htm_CXX_prefix.hxx
[ 67%] Building CXX object CMakeFiles/htmd.dir/src/htm/HtmServerMain.cpp.o
[ 68%] Building CXX object CMakeFiles/htm.dir/src/htm/HtmClientMain.cpp.o
[ 70%] Linking CXX static library libTerminalCommon.a
[ 70%] Built target TerminalCommon
[ 72%] Generating CXX prefix source cotire/etterminal_CXX_prefix.cxx
[ 73%] Generating CXX prefix source cotire/et_CXX_prefix.cxx
[ 73%] Generating CXX prefix source cotire/etserver_CXX_prefix.cxx
[ 75%] Generating CXX prefix source cotire/et-test_CXX_prefix.cxx
[ 77%] Generating CXX prefix header cotire/etterminal_CXX_prefix.hxx
[ 77%] Generating CXX prefix header cotire/et_CXX_prefix.hxx
[ 78%] Generating CXX prefix header cotire/etserver_CXX_prefix.hxx
[ 78%] Generating CXX prefix header cotire/et-test_CXX_prefix.hxx
[ 80%] Building CXX precompiled header cotire/et-test_CXX_prefix.hxx.pch
[ 80%] Building CXX object CMakeFiles/etterminal.dir/src/terminal/TerminalMain.cpp.o
[ 83%] Building CXX object CMakeFiles/etserver.dir/src/terminal/TerminalServerMain.cpp.o
[ 83%] Building CXX object CMakeFiles/et.dir/src/terminal/TerminalClientMain.cpp.o
[ 85%] Linking CXX executable htmd
[ 86%] Linking CXX executable htm
[ 86%] Built target htmd
[ 86%] Built target htm
[ 90%] Building CXX object CMakeFiles/et-test.dir/test/ConnectionTest.cpp.o
[ 90%] Building CXX object CMakeFiles/et-test.dir/test/BackedTest.cpp.o
[ 90%] Building CXX object CMakeFiles/et-test.dir/test/CryptoHandlerTest.cpp.o
[ 91%] Building CXX object CMakeFiles/et-test.dir/test/JumphostTest.cpp.o
[ 93%] Building CXX object CMakeFiles/et-test.dir/test/TerminalTest.cpp.o
[ 95%] Building CXX object CMakeFiles/et-test.dir/test/Main.cpp.o
[ 96%] Linking CXX executable etterminal
[ 98%] Linking CXX executable et
[100%] Linking CXX executable etserver
[100%] Built target etterminal
[100%] Linking CXX executable et-test
[100%] Built target et
[100%] Built target etserver
[100%] Built target et-test
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ ./et pudly
rgb .549
                 _ _
 _ __  _   _  __| | |_   _
| '_ \| | | |/ _` | | | | |
| |_) | |_| | (_| | | |_| |
| .__/ \__,_|\__,_|_|\__, |
|_|                  |___/
 __  __                               ___    _       _
|  \/  | ___  ___  ___ _____      __ |_ _|__| | __ _| |__   ___
| |\/| |/ _ \/ __|/ __/ _ \ \ /\ / /  | |/ _` |/ _` | '_ \ / _ \
| |  | | (_) \__ \ (_| (_) \ V  V /   | | (_| | (_| | | | | (_) |
|_|  |_|\___/|___/\___\___/ \_/\_( ) |___\__,_|\__,_|_| |_|\___/
                                 |/

       .-.      Light rain
      (   ).    +57(55) °F
     (___(__)   ↗ 9 mph
      ‘ ‘ ‘ ‘   8 mi
     ‘ ‘ ‘ ‘    0.0 in
(3)gentry@pudly ~ $ exit
logout
Session terminated

(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ psg etserver
root             1288   0.1  0.1  4359512  13904 s000  S+    3:37AM   0:00.78 ./etserver
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ kill 1288
-bash: kill: (1288) - Operation not permitted
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ sudo kill 1288
Password:
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ ./etserver
Abort trap: 6
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ psg etserver
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $ ./etserver
Abort trap: 6
(3)gentry@beast disable_telemetry_on_mojave:~/EternalTerminal/build $
```

**Comment by MisterTea** [comment]
Great!  Please update the other PR now that Mojave builds (we still need the ninja-build update)

File: src/base/Headers.hpp (hunk 1 above)
Comment (MisterTea) [reply src/base/Headers.hpp:166]: Yep

**Pull request merged**